src/main/java/io/supertokens/session/accessToken/AccessToken.java

```diff
@@ -155,12 +156,18 @@ public static AccessTokenInfo getInfoFromAccessToken(AppIdentifier appIdentifier
     }
 
     @TestOnly
-    public static AccessTokenInfo getInfoFromAccessTokenWithoutVerifying(@Nonnull String token) {
-        return getInfoFromAccessTokenWithoutVerifying(new AppIdentifier(null, null), token);
+    public static AccessTokenInfo getInfoFromAccessTokenWithoutVerifying(Main main, @Nonnull String token) {
+        try {
+            return getInfoFromAccessTokenWithoutVerifying(
+                    new AppIdentifier(null, null), main, token);
+        } catch (TenantOrAppNotFoundException e) {
+            throw new IllegalStateException(e);
+        }
     }
 
```

Comment:
In the AccessTokenPayload, we store the tenantId + another field called `appIdHash: hash(connectionUrl + appid)`

Whenever we decode / fetch info from the accessToken, we check if the hash of the input appIdentifier is the same as the hash in the access token payload. If not, we throw a badpermissionexception

Reply:
Done, throwing unauthorized exception (like it was doing before)